### ISSUE TITLE
About using a new dataset

### ISSUE THREAD

**Issue by SnycradJuice** [wgcban/ChangeFormer#40]
Hi, I found a new dataset called Landsat-SCD, which came from this paper[A transformer-based Siamese network and an open optical dataset for semantic change detection of remote sensing images](https://www.tandfonline.com/doi/full/10.1080/17538947.2022.2111470)
This dataset can be downloaded at [figshare](https://figshare.com/articles/figure/Landsat-SCD_dataset_zip/19946135/1)
Its structure is similar to LEVIR-CD, while the size is 416*416 and the label includes 10 classes, I wondered what should I do to run your model on this new dataset.
Thank you for your work.

**Comment by wgcban** [comment]
Hi @SnycradJuice 
You may follow these steps:

- For training, you can generate overlapping patches of size 256x256 with 128 pixels overlap. This way, you will have more training data.
- Next, ensure that the ground truth png file contains class numbers (0,1,2,3) when read.
- Specify the number of classes=4 in the script file by adding ``--n_class=4`` here: https://github.com/wgcban/ChangeFormer/blob/9025e26417cf8f10f29a48f34a05758498216465/scripts/run_ChangeFormer_WHU.sh#L37

**Comment by SnycradJuice** [comment]
Thanks for your help. : )

- Considering the dataset got 8000+ pairs of images,  I decided to resize them to fit the input size and then I found the method was given in your code :https://github.com/wgcban/ChangeFormer/blob/9025e26417cf8f10f29a48f34a05758498216465/datasets/data_utils.py#L61
- And I checked the value of GT png file, it's from 0 to 9, then I added  `--n_class` to sh file but still waiting for the end of the LEVIR-CD training.

**Comment by wgcban** [comment]
@SnycradJuice 
You can use the pre-trained model provided on the LEVIR-CD dataset unless you want to train it on your own.

**Comment by SnycradJuice** [comment]
Thank you for your help.
I'm running the Landsat-SCD dataset but the score is vey low, as the mf1 only reached 0.3864 at the 10th epoch, while the LEVIR-CD can reach 0.8857. Was it because the pretrained model doesn't fit the new dataset? I' m confused  @wgcban

**Comment by wgcban** [comment]
@SnycradJuice This is NOT usual. Have you visualize the predictions from the network after each epoch? It could be a bug in your implementation or we need to tune a hyperparamters a bit as this ismulti-class classification? Have you experimented with the different learning rate? 
Also try to check IOU scores for each class separately. mf1 means mean iou for class 1. What about other classes?

**Comment by SnycradJuice** [comment]
Well, after about 100 epochs, the score became normal, its f1 of ten classes is close to the paper.
Thank you!

**Comment by taemin6697** [comment]
@SnycradJuice 
I think I trained it on a new dataset of multi-class, can you tell me how you did it?
Or was it trained from scratch?
print("Initializing backbone weights from: " + self.args.pretrain)
         self.net_G.load_state_dict(torch.load(self.args.pretrain), strict=False)
         self.net_G.to(self.device)
         self.net_G.eval()
I wonder if it was trained using these pretrain weights.

**Comment by SnycradJuice** [comment]
Still used the pretrained_changeformer.pt

**Comment by SnycradJuice** [comment]
> @SnycradJuice I think I trained it on a new dataset of multi-class, can you tell me how you did it? Or was it trained from scratch? print("Initializing backbone weights from: " + self.args.pretrain) self.net_G.load_state_dict(torch.load(self.args.pretrain), strict=False) self.net_G.to(self.device) self.net_G.eval() I wonder if it was trained using these pretrain weights.

Still used the pretrained_changeformer.pt

**Comment by wangqingl** [comment]
> Hi, I found a new dataset called Landsat-SCD, which came from this paper[A transformer-based Siamese network and an open optical dataset for semantic change detection of remote sensing images](https://www.tandfonline.com/doi/full/10.1080/17538947.2022.2111470) This dataset can be downloaded at [figshare](https://figshare.com/articles/figure/Landsat-SCD_dataset_zip/19946135/1) Its structure is similar to LEVIR-CD, while the size is 416*416 and the label includes 10 classes, I wondered what should I do to run your model on this new dataset. Thank you for your work.

Hi, Why is the label of the dataset I downloaded a single channel image, not a three channel image, and all of it is black？Can you share me yours?Thank you.
![image](https://github.com/wgcban/ChangeFormer/assets/102707449/840caccf-993c-4cee-83d8-801f70978f35)

**Comment by wangqingl** [comment]
> Well, after about 100 epochs, the score became normal, its f1 of ten classes is close to the paper. Thank you for your advice!

Hello, I would like to know if your experiment on the dataset Landsat SCD using ChangeFormer yielded the same results as described in the paper proposing this dataset？
